src/sentry/reprocessing2.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import absolute_import
+
+import uuid
+import hashlib
+import logging
+import sentry_sdk
+
+from django.conf import settings
+
+from sentry import nodestore, features, eventstore
+from sentry.attachments import CachedAttachment, attachment_cache
+from sentry.models import EventAttachment
+from sentry.utils.cache import cache_key_for_event
+from sentry.eventstore.processing import event_processing_store
+
+logger = logging.getLogger("sentry.reprocessing")
+
+
+def _generate_unprocessed_event_node_id(project_id, event_id):
+    return hashlib.md5(
+        u"{}:{}:unprocessed".format(project_id, event_id).encode("utf-8")
+    ).hexdigest()
+
+
+def save_unprocessed_event(project, event_id):
+    """
+    Move event from event_processing_store into nodestore. Only call if event
+    has outcome=accepted.
+    """
+    if not features.has("projects:reprocessing-v2", project, actor=None):
+        return
+
+    with sentry_sdk.start_span(
+        op="sentry.reprocessing2.save_unprocessed_event.get_unprocessed_event"
+    ):
+        data = event_processing_store.get(
+            cache_key_for_event({"project": project.id, "event_id": event_id}), unprocessed=True
+        )
+        if data is None:
+            return
+
+    with sentry_sdk.start_span(op="sentry.reprocessing2.save_unprocessed_event.set_nodestore"):
+        node_id = _generate_unprocessed_event_node_id(project_id=project.id, event_id=event_id)
+        nodestore.set(node_id, data)
```

Comment:
_nit_: Current node store writes are located in EventManager code. Can we move this over to consolidate, and just leave the reprocessing logic here? If you prefer to keep it here, that's fine with me, too.

Reply:
I'll keep it here to keep `_generate_unprocessed_event_node_id` private